sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -697,6 +704,8 @@ public Query visitFunction(Function function, Context context) {
             if (fieldIgnored(function, context)) {
                 return Queries.newMatchAllQuery();
             }
+            unsupportedFields(function, context);
```

Comment:
maybe rename the method to make it a bit clearer that this method does validation and throws an exception